client/my-sites/promote-post-i2/components/posts-list/header.tsx

```diff
@@ -5,46 +5,62 @@ type PostsListHeaderColumn = {
 	title: string | null;
 };
 
-export default function PostsListHeader() {
-	const columns: Array< PostsListHeaderColumn > = [
-		{
-			id: 'data',
-			title: translate( 'Post' ),
-		},
-		{
-			id: 'type',
-			title: translate( 'Type' ),
-		},
-		{
+type PostsListHeaderProps = {
+	postType: 'product' | 'post' | 'page';
+};
+
+export default function PostsListHeader( { postType: postType }: PostsListHeaderProps ) {
+	const getHeaderColumns = (): Array< PostsListHeaderColumn > => {
+		const columns = [
+			{
+				id: 'data',
+				title: translate( 'Post' ),
+			},
+			{
+				id: 'type',
+				title: translate( 'Type' ),
+			},
+		];
+
+		if ( postType === 'product' ) {
```

Comment:
We need to check here the same condition we use in the Table body; if not, it may be possible for the table to have more headers than rows. 

Here, instead of checking against the postType, you need to check against the WooStore feature flag and the filter post type. That way, both checks will match, and the table will always have the same number of columns in the header and body.

Reply:
maybe the naming is not correct but this `postType` is the filterType that comes from the` posts-list `file

Reply:
Also the condition to execute this is:

```
{ isRunningInWooStore && type === 'product' ? (
				<PostsListHeader postType={ type } />
			) : (
				<PostsListHeader postType="post" />
			) }
```

so if there's an `All` type filter it will show as a `post` type header

Reply:
Fixed